File: frontend/projects/domain/src/lib/quote-input.ts
import { ResolvedLocation } from './resolved-location';
export interface QuoteInput {
  service: string;
  startsAt: string;
  endsAt: string;
  locations: {
    location: ResolvedLocation;
    parkingAmount: string;
    studioId: string | null;
    studioHours: string;
  }[];
  assistantCount: number;
  equipmentUnits: number;
  lunchCount: number;
  code: string;
  photographerId: string | null;
  inputRevision: number;
}

File: backend/src/QuinntyneBrownStudio.Domain/Models/QuoteInput.cs
using QuinntyneBrownStudio.Domain.Enums;

namespace QuinntyneBrownStudio.Domain.Models;

public sealed class QuoteInput
{
    public ServiceKind Service { get; set; }
    public DateTimeOffset StartsAt { get; set; }
    public DateTimeOffset EndsAt { get; set; }
    public QuoteLocation[] Locations { get; set; } = [];
    public int AssistantCount { get; set; }
    public int EquipmentUnits { get; set; }
    public int LunchCount { get; set; }
    public string? Code { get; set; }
    public Guid? PhotographerId { get; set; }
    public long InputRevision { get; set; }
}
